core/src/main/java/hudson/PluginManager.java

```diff
@@ -1086,8 +1086,7 @@ protected void copyBundledPlugin(URL src, String fileName) throws IOException {
                 }
             } finally {
                 Util.closeAndLogFailures(in, LOGGER, PluginWrapper.MANIFEST_FILENAME, bundledJpi.toString());
-                if (cl instanceof Closeable)
-                    ((Closeable)cl).close();
+                ((Closeable)cl).close();
```

Comment:
You could remove this line entirely if you move line 1079 into a `try (URLClassLoader cl = ...) { ... }` block.